src/sparse/impl/KokkosSparse_spgemm_mkl_impl.hpp

```diff
@@ -47,634 +47,383 @@
 
 #ifdef KOKKOSKERNELS_ENABLE_TPL_MKL
 #include "mkl_spblas.h"
-#include "mkl.h"
 #endif
 
-#include "KokkosKernels_Utils.hpp"
-#include <Kokkos_Concepts.hpp>
-
 namespace KokkosSparse {
-
 namespace Impl {
 
-template <typename KernelHandle, typename in_row_index_view_type,
-          typename in_nonzero_index_view_type, typename bin_row_index_view_type,
-          typename bin_nonzero_index_view_type,
-          typename cin_row_index_view_type>
-void mkl_symbolic(KernelHandle *handle, typename KernelHandle::nnz_lno_t m,
-                  typename KernelHandle::nnz_lno_t n,
-                  typename KernelHandle::nnz_lno_t k,
-                  in_row_index_view_type row_mapA,
-                  in_nonzero_index_view_type entriesA,
-
-                  bool transposeA, bin_row_index_view_type row_mapB,
-                  bin_nonzero_index_view_type entriesB, bool transposeB,
-                  cin_row_index_view_type row_mapC, bool verbose = false) {
 #ifdef KOKKOSKERNELS_ENABLE_TPL_MKL
 
-  typedef typename KernelHandle::nnz_lno_t idx;
-  typedef typename KernelHandle::size_type size_type;
-
-  typedef typename KernelHandle::HandleTempMemorySpace HandleTempMemorySpace;
-  typedef typename Kokkos::View<int *, HandleTempMemorySpace>
-      int_temp_work_view_t;
+inline void mkl_call(sparse_status_t result, const char *err_msg) {
```

Comment:
It might be time to create `src/common/KokkosKernels_SparseUtils_mkl.hpp` for this type of utility function. Also there is a similar function in src/impl/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp' called `mkl_safe_call` as well as `mode_kk_to_mkl` which should be consolidated.

Reply:
I've moved MKL functions to `src/common/KokkosKernels_SparseUtils_mkl.hpp` on 0150764529468e96c7695c5ff6cea801346f0461